Backend/src/test/java/ktsnvt/tim1/controllers/LocationControllerIntegrationTests.java

```diff
@@ -401,21 +401,103 @@ void createSeatGroup_locationDoesNotExist_errorMessageReturned() {
         newDTO.setxCoordinate(3.0);
         newDTO.setyCoordinate(4.0);
         newDTO.setName("Group1");
+        newDTO.setAngle(30.0);
 
         HttpEntity<SeatGroupDTO> entity = new HttpEntity<>(newDTO);
 
         ResponseEntity<String> result = testRestTemplate
-                .exchange(createURLWithPort("/locations/31/seat-groups"), HttpMethod.POST, entity, String.class);
+                .exchange("/locations/31/seat-groups", HttpMethod.POST, entity, String.class);
 
         assertEquals(HttpStatus.NOT_FOUND, result.getStatusCode());
         assertEquals("Location not found", result.getBody());
         assertFalse(l.isPresent());
     }
 
+    @Test
+    void getLocationsOptions_locationsOptionsReturned() {
+        int locationOptionsCount = 31;
+
+        ParameterizedTypeReference<List<LocationOptionDTO>> responseType =
+                new ParameterizedTypeReference<List<LocationOptionDTO>>() {
+                };
+
+        ResponseEntity<List<LocationOptionDTO>> result = testRestTemplate
+                .exchange("/locations/options", HttpMethod.GET, null, responseType);
+
+        assertNotNull(result);
+        assertEquals(HttpStatus.OK, result.getStatusCode());
+        assertNotNull(result.getBody());
+        assertEquals(locationOptionsCount, result.getBody().size());
+    }
+
+    @Transactional
```

Comment:
Remove @Transactional and @Rollback from controller methods because there is no way of rolling back the transactions triggered by the requests made through the RestTemplate in a test method